app/utils/navigators/IndividualNavigator.scala

```diff
@@ -75,40 +84,86 @@ class IndividualNavigator @Inject()(val dataCacheConnector: UserAnswersCacheConn
   }
 
   def addressYearsRoutes(answers: UserAnswers): Option[NavigateTo] = {
-    answers.get(IndividualAddressYearsId) match {
-      case Some(AddressYears.UnderAYear) =>
+    (answers.get(IndividualAddressYearsId), answers.get(AreYouInUKId)) match {
```

Comment:
This should use a feature toggle. case _ could be case None which is not a valid data state with Non-UK switched on. We would want to fail fast in that scenario. This applies to all uses of (answers.get(???), answers.get(AreYouInUKId)) match ...